src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/set_global_properties_test.cc

```diff
@@ -1262,6 +1262,205 @@ TEST_F(SetGlobalPropertiesRequestTest,
   command->Run();
 }
 
+TEST_F(SetGlobalPropertiesRequestTest,
+       Run_InvalidCustomizableKeys_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject customizable_keys(smart_objects::SmartType_Array);
+  customizable_keys[0] = "%";
+  customizable_keys[1] = "\\n";
+  customizable_keys[2] = " ";
+  (*msg)[am::strings::msg_params][am::strings::keyboard_properties]
+        [am::hmi_request::customize_keys] = customizable_keys;
+
+  std::shared_ptr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(
+    SetGlobalPropertiesRequestTest,
+    Run_DisallowedNumberOfCustomizableKeysAndLayoutFromRequest_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject customizable_keys(smart_objects::SmartType_Array);
+  customizable_keys[0] = "%";
+  customizable_keys[1] = "@";
+  customizable_keys[2] = "&";
+  (*msg)[am::strings::msg_params][am::strings::keyboard_properties]
+        [am::hmi_request::customize_keys] = customizable_keys;
+  (*msg)[am::strings::msg_params][am::strings::keyboard_properties]
+        [am::hmi_request::keyboard_layout] =
+            hmi_apis::Common_KeyboardLayout::QWERTZ;
+
+  const int num_allowed_keys = 0;
+  auto display_capabilities =
+      std::make_shared<SmartObject>(smart_objects::SmartType_Map);
+  auto& configurable_keyboards =
+      (*display_capabilities)[0][am::strings::window_capabilities][0]
+                             [am::hmi_response::keyboard_capabilities]
+                             [am::hmi_response::configurable_keys];
+  configurable_keyboards[0][am::hmi_request::keyboard_layout] =
+      hmi_apis::Common_KeyboardLayout::QWERTZ;
+  configurable_keyboards[0][am::hmi_response::num_configurable_keys] =
+      num_allowed_keys;
```

Comment:
@ychernysheva good candidate for a separate function as well - can accept two parameters (num_allowed_keys, layout)
returns ptr to display capabilities

Reply:
Done in f10bc21